proj/src/main/webapp/signup.js

```diff
@@ -8,30 +8,50 @@ function initSignUp(){
     location.getLocation();
 }
 
+// if(btnLogin){
+//         btnLogin.addEventListener("click", e => {
+//             const emailVal = txtEmail.value;
+//             const passVal = txtPassword.value;
+
+//             // Initialize auth object
+//             const auth = firebase.auth();
+
+//             const promise = auth.signInWithEmailAndPassword(emailVal, passVal).then(function(){
+//                 window.location.replace("chat.html");
+//             });
+//             promise.catch(e => alert(e.message));
+//         });
+//     }
+
 function initSignUpButtons(){
     const btnSignUp = document.getElementById("btnSignUp");
     // Add sign up event
     if(btnSignUp){
         btnSignUp.addEventListener("click", async function () {
-            // Elements of sign up container
-            const fname = document.getElementById("fname");
-            const lname = document.getElementById("lname");
-            const txtEmail = document.getElementById("email");
-            const txtPassword = document.getElementById("pass");
-            const tagStr = document.getElementById("tags");
-            const isLoc = document.getElementById("loc");
-            var lat;
-            var long;
-            if(isLoc.checked){
-                const coords = location.getLatLong();
-                lat = coords[0];
-                long = coords[1]; 
+            try {
+                // Elements of sign up container
+                const fname = document.getElementById("fname");
+                const lname = document.getElementById("lname");
+                const txtEmail = document.getElementById("email");
+                const txtPassword = document.getElementById("pass");
+                const tagStr = document.getElementById("tags");
+                const isLoc = document.getElementById("loc");
+                var lat;
+                var long;
+                if(isLoc.checked){
+                    const coords = location.getLatLong();
+                    lat = coords[0];
+                    long = coords[1]; 
+                }
+                else{
+                    lat = 999;
+                    long = 999;
+                }
+                await signUp(fname.value, lname.value, txtEmail.value, txtPassword.value, tagStr.value, lat, long);
             }
-            else{
-                lat = 999;
-                long = 999;
+            catch (e) {
+                alert(e.message);
```

Comment:
Does this need to be wrapped in a try/catch? What happens if it isn't? Can the error catching happen closer to where the error actually gets thrown.

Reply:
This smells like debugging code to me.  Is the error message user-oriented?  (That is, could the user take action based on this message?)  If not, it would be better to catch it in the most specific scope possible, and make some graceful UI change rather than just throwing up an alert.

Reply:
the alert is target for the user so they can see what information is missing or incorrect. and i think i understand what you guys mean, i am wrapping the try catch around the signup function